core/src/com/unciv/ui/overviewscreen/ReligionOverviewTable.kt

```diff
@@ -0,0 +1,117 @@
+package com.unciv.ui.overviewscreen
+
+import com.badlogic.gdx.graphics.Camera
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.scenes.scene2d.ui.Button
+import com.badlogic.gdx.scenes.scene2d.ui.Label
+import com.badlogic.gdx.scenes.scene2d.ui.Table
+import com.unciv.logic.civilization.CivilizationInfo
+import com.unciv.models.Religion
+import com.unciv.models.ruleset.Belief
+import com.unciv.models.translations.tr
+import com.unciv.ui.utils.*
+import kotlin.math.min
+
+class ReligionOverviewTable(
+    private val viewingPlayer: CivilizationInfo,
+    private val overviewScreen: EmpireOverviewScreen
+): Table() {
+    
+    val gameInfo = viewingPlayer.gameInfo
+    private val topButtons = Table(CameraStageBaseScreen.skin)
+    private val topButtonLabel = "Click an icon to see the stats of this religion".tr().toLabel()
+    private val statsTable = Table(CameraStageBaseScreen.skin)
+    private val beliefsTable = Table(CameraStageBaseScreen.skin)
+    private var selectedReligion: String? = null
+    
+    init {
+        draw(null)
+    }
+    
+    private fun draw(religion: Religion?) {
+        topButtons.clear()
+        topButtonLabel.clear()
+        statsTable.clear()
+        beliefsTable.clear()
+        clear()
+
+        addReligionButtons()
+        if (religion != null) loadReligion(religion)
+        add(topButtons).pad(5f).row()
+        add(topButtonLabel).pad(5f)
+        addSeparator()
+        add(statsTable).pad(5f).row()
+        add(beliefsTable).pad(5f)
```

Comment:
Why are you resetting everything and not just resetting the parts in loadReligion?